community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/IndexingService.java

```diff
@@ -272,25 +270,35 @@ public void start() throws Exception
             dropRecoveringIndexes( indexMap, rebuildingDescriptors.iterator() );
 
             // Rebuild indexes by recreating and repopulating them
-            if ( !rebuildingDescriptors.isEmpty() )
+            PrimitiveLongObjectMap<RebuildingIndexDescriptor> rebuildingNodeDescriptors = Primitive.longObjectMap();
+            PrimitiveLongObjectMap<RebuildingIndexDescriptor> rebuildingRelationshipDescriptors = Primitive.longObjectMap();
+            PrimitiveLongIterator iterator = rebuildingDescriptors.iterator();
+            while ( iterator.hasNext() )
+            {
+                long indexId = iterator.next();
+                RebuildingIndexDescriptor descriptor = rebuildingDescriptors.get( indexId );
+                if ( descriptor.getIndexDescriptor().schema().entityType() == EntityType.NODE )
+                {
+                    rebuildingNodeDescriptors.put( indexId, descriptor );
+                }
+                else
+                {
+                    rebuildingRelationshipDescriptors.put( indexId, descriptor );
+                }
+            }
+
+            if ( !rebuildingNodeDescriptors.isEmpty() )
```

Comment:
i'm sure that starting those 2 jobs as they are now will not gonna play well since each of those will gonna try to perform multi-threaded index population with tons of threads fighting with each other for no reason

Reply:
In what sense will they fight? They scan different stores and apply updates to different indexes?